packages/cli/src/handlers/oauthLogin.ts

```diff
@@ -0,0 +1,374 @@
+import {
+    AuthorizationError,
+    CreatePersonalAccessToken,
+    formatDate,
+    oauthPageStyles,
+} from '@lightdash/common';
+import { exec } from 'child_process';
+import * as crypto from 'crypto';
+import * as http from 'http';
+import { createServer } from 'net';
+import fetch from 'node-fetch';
+import { URL } from 'url';
+import { promisify } from 'util';
+import GlobalState from '../globalState';
+import * as styles from '../styles';
+
+// Shared CSS styles for OAuth response pages - matching Lightdash login design
+
+const execAsync = promisify(exec);
+
+// PKCE helper functions
+const generateCodeVerifier = (): string =>
+    crypto.randomBytes(32).toString('base64url');
+
+const generateCodeChallenge = (verifier: string): string =>
+    crypto.createHash('sha256').update(verifier).digest('base64url');
+
+const generatePersonalAccessToken = async (
```

Comment:
my assumption was that the OAuth flow would return the personal access token AS the access token directly, is that not possible?